### ISSUE TITLE
RuntimeError: Timeout context manager should be used inside a task

### ISSUE THREAD

**Issue by MarkEdmondson1234** [googleapis/langchain-google-alloydb-pg-python#160]
I have for a few months been trying to mitigate this error that occurs occasionally when a customer is using the AlloyDB Langchain connection.  It looks like any error such as not being able to connect occasionally then causes the Cloud Run container to get stuck on repeat, which I can only fix by rebooting the container.

The bots keep saying its because "the aiohttp library’s timeout context manager is being used incorrectly. Specifically, it’s being used outside of an asyncio task." but I don't know if thats any help.  My attempts to catch the error do not work, so I think any change needs to be within this module itself, hence the issue.

#### Environment details

  - OS type and version: Cloud Run - Docker derived FROM python:3.10-slim
  - Python version: 3.10
  - `langchain-google-alloydb-pg` version: 0.2.2

#### Steps to reproduce

  1. Have a connection error drop occasionally - hard to reproduce consistently

#### Code example

I've tried async and sync versions but it doesn't seem to help.

```python
...
        vectorstore = AlloyDBVectorStore.create_sync(
                engine=engine,
                table_name=table_name,
                embedding_service=embeddings,
                metadata_columns=["source", "docstore_doc_id"]
            )
        retriever = vectorstore.as_retriever()
        docs = retriever.invoke(query)
```

#### Stack trace
The container loops with this stack trace, but note it can be several different root causes such as not connecting correctly or a syntax error.  My last example is this: 

```
google.cloud.alloydb.connector.instance - INFO - ['projects/project_id/locations/europe-west1/clusters/my-cluster-20240401/instances/primary-instance-1']: An error occurred while performing refresh. Scheduling another refresh attempt immediately
```

Then it gets stuck doing this:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/google/cloud/alloydb/connector/instance.py", line 195, in _refresh_operation
    refresh_result = await refresh_task
  File "/usr/local/lib/python3.10/site-packages/google/cloud/alloydb/connector/instance.py", line 152, in _perform_refresh
    ip_addr, certs = await asyncio.gather(metadata_task, certs_task)
  File "/usr/local/lib/python3.10/site-packages/google/cloud/alloydb/connector/client.py", line 129, in _get_metadata
    resp = await self._client.get(url, headers=headers, raise_for_status=True)
  File "/usr/local/lib/python3.10/site-packages/aiohttp/client.py", line 507, in _request
    with timer:
  File "/usr/local/lib/python3.10/site-packages/aiohttp/helpers.py", line 715, in __enter__
    raise RuntimeError(
RuntimeError: Timeout context manager should be used inside a task
```

Hope you can help, or let me know if there is some way I can mitigate the call itself.  My main concern is not needing to reboot every few days due to this error.

**Comment by averikitsch** [comment]
@jackwotherspoon This looks like an issue with the connector trying to refresh/reconnect. Do you have an idea if this is an issue for the integration or for how the application code is using the engine?

**Comment by jackwotherspoon** [comment]
@MarkEdmondson1234 @averikitsch The AlloyDB Python Connector had a release today where I added a new `refresh_strategy` argument to the Connector that when set to `"lazy"` will have the connector use a lazy refresh strategy instead of the default `"background"` which is to run background refreshes to get a fresh client cert for the SSL/TLS connection.

The lazy refresh should be used in serverless environments because background refreshes when serverless environments scale to zero and throttle the CPU will fail with errors like the one @MarkEdmondson1234 is seeing.

TLDR; @averikitsch we should update this package to the latest Python Connector version `google-cloud-alloydb-connector==1.2.0` and to use the lazy refresh strategy for the Connector.

https://github.com/googleapis/langchain-google-alloydb-pg-python/blob/01b4eb2525d9af62a47bebefa188776ca9f3f355/src/langchain_google_alloydb_pg/alloydb_engine.py#L170

```python
AsyncConnector(user_agent=USER_AGENT, refresh_strategy="lazy") 
```

Hope this helps, I will work on the PR to fix this library in the next few days.

**Comment by averikitsch** [comment]
Thank you @jackwotherspoon !

**Comment by MarkEdmondson1234** [comment]
Many thanks for quick resolution! @jackwotherspoon

**Comment by jamiedevoteam** [comment]
Hi @MarkEdmondson1234, I was wondering if you still encounter the above mentioned issue after the update? Because I still encounter this issue despite updating my packages (and by default after updating, it should be refresh=lazy from the code). Did it solve your issue?

**Comment by MarkEdmondson1234** [comment]
Hi @jamiedevoteam (do I know you? I worked at Devoteam Denmark last year) 

I have not come across the issue since the update so yes I think it’s solved!

**Comment by jamiedevoteam** [comment]
Hi @MarkEdmondson1234, probably not, I'm from Devoteam Netherlands :). Can I know the steps you took to resolve the issue? Just to be sure that I did not miss out on anything. Are you also using async calls?

**Comment by MarkEdmondson1234** [comment]
I know a few folks there, hi :) I just let the cloud run docket build to roll on to the next version of this pip install, and have not seen it since. Using async and sync. No code changes.
